x/leverage/keeper/keeper.go

```diff
@@ -584,3 +584,26 @@ func (k Keeper) LiquidationParams(ctx sdk.Context, reward string, borrowed, limi
 
 	return liquidationIncentive, closeFactor, nil
 }
+
+// Iterate through all the keys from that prefix and then calls
+// the iterate function with the key and value if this function
+// returns an error, it stops the for loop
+func (k Keeper) Iterate(ctx sdk.Context, prefix []byte, iterate func(key, val []byte) error) error {
```

Comment:
We might want to make this function name lowercase `iterate` so it's not part of the leverage keeper's exported interface, just in case. Thoughts, @alexanderbez ?

Good idea on making this function and the associated helpers @RafilxTenfen , should make the rest of the code simpler.

Reply:
Agreed.